Tests/NIOCoreTests/AsyncChannel/AsynChannelUnixDomainSocketTests.swift

```diff
@@ -0,0 +1,171 @@
+                    try await outbound.write("Hello")
+
+                    var inboundIterator = inbound.makeAsyncIterator()
+                    guard let messageA = try await inboundIterator.next() else { return }
+                    print("C: Did receive '\(messageA)'")
+                    clientDidReceive.confirm()
+                    #expect(messageA == "Hello")
+
+                    try await outbound.write("QUIT")
+                }
+            }
+
+            for try await connectionChannel in inbound {
+                group.addTask {
+                    print("Handling new connection")
+                    await handleConnection(
+                        channel: connectionChannel,
+                        serverDidReceive: serverDidReceive
+                    )
+                    print("Done handling connection")
+                }
+                break
+            }
+        }
+    }
+}
+
+private func makeServerChannel(
+    path: String
+) async throws -> NIOAsyncChannel<NIOAsyncChannel<String, String>, Never> {
+    try await ServerBootstrap(
+        group: NIOSingletons.posixEventLoopGroup
+    ).bind(
+        unixDomainSocketPath: path,
+        cleanupExistingSocketFile: true,
+        serverBackPressureStrategy: nil
+    ) { childChannel in
+        childChannel.eventLoop.makeCompletedFuture {
+            try childChannel.pipeline.syncOperations.addHandler(ByteToMessageHandler(NewlineDelimiterCoder()))
+            try childChannel.pipeline.syncOperations.addHandler(MessageToByteHandler(NewlineDelimiterCoder()))
+            return try NIOAsyncChannel<String, String>(
+                wrappingChannelSynchronously: childChannel
+            )
+        }
+    }
+}
+
+private func makeClientChannel(
+    path: String
+) async throws -> NIOAsyncChannel<String, String> {
+    try await ClientBootstrap(group: NIOSingletons.posixEventLoopGroup)
+        .connect(unixDomainSocketPath: path)
+        .flatMap { channel in
+            channel.eventLoop.makeCompletedFuture {
+                try channel.pipeline.syncOperations.addHandler(ByteToMessageHandler(NewlineDelimiterCoder()))
+                try channel.pipeline.syncOperations.addHandler(MessageToByteHandler(NewlineDelimiterCoder()))
+                return try NIOAsyncChannel<String, String>(wrappingChannelSynchronously: channel)
+            }
+        }
+        .get()
+}
+
+private func handleConnection(
+    channel: NIOAsyncChannel<String, String>,
+    serverDidReceive: Confirmation
+) async {
+    do {
+        print("S: New channel")
+        try await channel.executeThenClose { inbound, outbound in
+            for try await message in inbound {
+                print("S: Did receive '\(message)'")
+                guard message != "QUIT" else { return }
+                serverDidReceive.confirm()
+                try await outbound.write(message)
+            }
+            print("S: Bye")
+        }
+    } catch {
```

Comment:
Can we just make `handleConnection` throwing and propagate that error up rather than suppressing it?